Core/Source/MVVM/ViewModelUserEvents.swift

```diff
@@ -0,0 +1,206 @@
+import Foundation
+
+/// Event types that view models may handle, typically sent from the view layer.
+public enum ViewModelUserEvent {
+    /// On mouse-supporting platforms, represents a click by the user.
+    case click
+
+    /// Represents the user typing a key with modifier flags.
+    case keyDown(EventKeyCode, EventKeyModifierFlags)
+
+    /// Reprsents the user performing a long-press on the target view model.
+    case longPress
+
+    /// On mouse-supporting platforms, represents a secondary (right) click by the user.
+    case secondaryClick
+
+    /// On any platform, represents the target being selected (via mouse, programatically, or tap).
+    case select
+
+    /// On touch platforms, represents the target receiving a single tap.
+    case tap
+}
+
+// swiftlint:disable type_name
+
+/// Possible key code values for `NSEvent.keyCode` - taken from <HIToolbox/Events.h>.
+@objc
+public enum EventKeyCode: UInt16, RawRepresentable {
```

Comment:
could consider putting this at the end of the file since it's sorta the least interesting part (and long)